lib/segment/src/index/field_index/map_index.rs

```diff
@@ -271,14 +414,25 @@ impl PayloadFieldIndex for MapIndex<IntPayloadType> {
         MapIndex::flusher(self)
     }
 
-    fn filter(
-        &self,
-        condition: &FieldCondition,
-    ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + '_>> {
+    fn filter<'a>(
+        &'a self,
+        condition: &'a FieldCondition,
+    ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + 'a>> {
         match &condition.r#match {
             Some(Match::Value(MatchValue {
                 value: ValueVariants::Integer(integer),
             })) => Some(self.get_iterator(integer)),
+            Some(Match::Any(MatchAny {
+                any: AnyVariants::Integers(integers),
+            })) => Some(Box::new(
+                integers
+                    .iter()
+                    .flat_map(move |integer| self.get_iterator(integer))
```

Comment:
`move` is not required here:

```suggestion
                    .flat_map(|integer| self.get_iterator(integer))
```